plugin/src/main/java/org/opensearch/ml/action/models/UpdateModelTransportAction.java

```diff
@@ -102,66 +119,79 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<Update
+                                    if (hasPermission) {
+                                        updateRemoteOrTextEmbeddingModel(
+                                            modelId,
+                                            updateModelInput,
+                                            mlModel,
+                                            user,
+                                            wrappedListener,
+                                            isModelDeployed
+                                        );
                                     } else {
-                                        actionListener
+                                        wrappedListener
                                             .onFailure(
                                                 new OpenSearchStatusException(
-                                                    "ML Model "
-                                                        + modelId
-                                                        + " is in deploying or deployed state, please undeploy the models first!",
+                                                    "User doesn't have privilege to perform this operation on this model, model ID "
+                                                        + modelId,
                                                     RestStatus.FORBIDDEN
                                                 )
                                             );
                                     }
-                                } else {
-                                    actionListener
-                                        .onFailure(
-                                            new OpenSearchStatusException(
-                                                "User doesn't have privilege to perform this operation on this model, model ID " + modelId,
-                                                RestStatus.FORBIDDEN
-                                            )
-                                        );
-                                }
-                            }, exception -> {
-                                log.error("Permission denied: Unable to update the model with ID {}. Details: {}", modelId, exception);
-                                actionListener.onFailure(exception);
-                            }));
-                    }
+                                }, exception -> {
+                                    log.error("Permission denied: Unable to update the model with ID {}. Details: {}", modelId, exception);
+                                    wrappedListener.onFailure(exception);
+                                }));
+                        }
 
+                    } else {
+                        wrappedListener
+                            .onFailure(
+                                new MLValidationException(
+                                    "User doesn't have privilege to perform this operation on this function category: "
```

Comment:
This `else` block is for `if (functionName == TEXT_EMBEDDING || functionName == REMOTE) {` 

I don't see user privilege checking, why this error says `User doesn't have privilege`

Reply:
Then any suggested wording? What about `We don't support...`

Reply:
> We don't support.../... were not supported at this time.

Sounds good